RecoTracker/FinalTrackSelectors/plugins/TrackSelectorByRegion.cc

```
@@ -0,0 +1,166 @@
+      inputTrkRegionToken_ = consumes<edm::OwnVector<TrackingRegion>>(regionPSet.getParameter<edm::InputTag>("input"));
+      phiTollerance_ = regionPSet.getParameter<double>("phiTollerance");
+      edm::ParameterSet trackPSet = conf.getParameter<edm::ParameterSet>("TrackPSet");
+      minPt_ = trackPSet.getParameter<double>("minPt");
+      //      }
+
+      produces<reco::TrackCollection>();
+      // produces<edm::RefToBaseVector<T> >();
+      produces<std::vector<bool>>();
+    }
+
+    static void fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+      edm::ParameterSetDescription desc;
+      desc.add<edm::InputTag>("tracks", edm::InputTag("hltPixelTracks"));
+      edm::ParameterSetDescription tracks_par;
+      tracks_par.add<double>("minPt", 0.);
+      desc.add<edm::ParameterSetDescription>("TrackPSet", tracks_par);
+
+      edm::ParameterSetDescription region_par;
+      region_par.add<edm::InputTag>("input", edm::InputTag(""));
+      region_par.add<double>("phiTollerance", 1.);
+      desc.add<edm::ParameterSetDescription>("RegionPSet", region_par);
+      descriptions.add("trackSelectorByRegion", desc);
+    }
+
+  private:
+    using MaskCollection = std::vector<bool>;
+
+    void produce(edm::StreamID, edm::Event& iEvent, const edm::EventSetup&) const override {
+      // products
+      auto mask = std::make_unique<MaskCollection>();                  // mask w/ the same size of the input collection
+      auto output_tracks = std::make_unique<reco::TrackCollection>();  // selected output collection
+
+      edm::Handle<edm::OwnVector<TrackingRegion>> regionsHandle;
+      iEvent.getByToken(inputTrkRegionToken_, regionsHandle);
+
+      edm::Handle<reco::TrackCollection> tracksHandle;
+      iEvent.getByToken(tracksToken_, tracksHandle);
+
+      //	std::cout << "[TrackSelectorByRegion::produce] tracksHandle.isValid ? " << (tracksHandle.isValid() ? "YEAP" : "NOPE") << std::endl;
+      if (tracksHandle.isValid()) {
```

as in TrackSelectorByCandidateRegion: is there some intrinsic limitation in the HLT to miss writing a track collection in the event? If not, to avoid configuration problems it would be more appropriate to not check for validity and let the job fail if it's incorrectly configured